### ISSUE TITLE
Segmentation fault in DataDog agent

### ISSUE THREAD

**Issue by omb** [DataDog/datadog-agent#2802]
Datadog Agent v6.8.1

**Docker log for the agent**
```
[ TRACE ] 2018-12-18 08:21:35 INFO (trace.go:102) - flushed trace payload to the API, time:391.07429ms, size:2610 bytes
[ TRACE ] 2018-12-18 08:21:35 INFO (stats.go:265) - flushed stat payload; url: https://trace.agent.datadoghq.com, time:426.09175ms, size:1285 bytes
[ TRACE ] 2018-12-18 08:21:40 INFO (trace.go:102) - flushed trace payload to the API, time:336.876926ms, size:3761 bytes
[ AGENT ] panic: runtime error: invalid memory address or nil pointer dereference
[ AGENT ] [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1658562]
[ AGENT ] 
[ AGENT ] goroutine 109 [running]:
[ AGENT ] github.com/DataDog/datadog-agent/pkg/logs/config.(*LogSource).RemoveInput(0x0, 0xc420ffce69, 0x40)
[ AGENT ] 	/go/src/github.com/DataDog/datadog-agent/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/config/source.go:49 +0x22
[ AGENT ] github.com/DataDog/datadog-agent/pkg/logs/input/docker.(*Launcher).stopTailer(0xc42056ea20, 0xc420ffce69, 0x40)
[ AGENT ] 	/go/src/github.com/DataDog/datadog-agent/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/launcher.go:231 +0x9c
[ AGENT ] github.com/DataDog/datadog-agent/pkg/logs/input/docker.(*Launcher).run(0xc42056ea20)
[ AGENT ] 	/go/src/github.com/DataDog/datadog-agent/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/launcher.go:160 +0x4f0
[ AGENT ] created by github.com/DataDog/datadog-agent/pkg/logs/input/docker.(*Launcher).Start
[ AGENT ] 	/go/src/github.com/DataDog/datadog-agent/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/launcher.go:93 +0x3f
[ AGENT ] AGENT EXITED WITH CODE 2, SIGNAL 0, KILLING CONTAINER
[ AGENT ] starting agent
[ TRACE ] 2018-12-18 08:21:42 INFO (main.go:35) - received signal 15 (terminated)
[ TRACE ] 2018-12-18 08:21:42 INFO (agent.go:140) - exiting
[ TRACE ] 2018-12-18 08:21:42 INFO (concentrator.go:91) - exiting concentrator, computing remaining stats
[ TRACE ] 2018-12-18 08:21:42 INFO (trace.go:134) - exiting trace writer, flushing all remaining traces
[ TRACE ] 2018-12-18 08:21:42 INFO (trace.go:137) - Flushed. Exiting
[ AGENT ] AGENT EXITED WITH CODE 256, SIGNAL 15, KILLING CONTAINER
[PROCESS] process-agent exited with code 0, disabling
```

**Additional environment details (Operating System, Cloud provider, etc):**
ECS on AWS

**Comment by achntrl** [comment]
Hi @omb, thank you for reporting this issue. 

We will need to look into the logs of your agent(s). 

The best would be to open a support case ([see here](https://www.datadoghq.com/support/)) and send over a flare. The flare is a bundle of logs and configs that is forwarded to our support team for us to get more context around the issue.

You can create a flare by running: `agent flare <caseid>`

**Comment by omb** [comment]
Support case ticket id: 187904

**Comment by achntrl** [comment]
Hi @omb, we've just released the agent 6.8.2 that fixes this bug.

Thank you again for reporting the problem!
